interface/modules/custom_modules/oe-module-comlink-telehealth/src/Controller/TeleconferenceRoomController.php

```diff
@@ -0,0 +1,786 @@
+            }
+            if ($session['user_id'] != $user['id']) {
+                // throw an error
+                throw new AccessDeniedException("patients", "appt", "Cannot update appointment that belongs to different user");
+            }
+
+//            if (!$this->appointmentService->isValidAppointmentStatus($status)) {
+            if (!$this->isValidAppointmentStatus($status)) {
+                throw new InvalidArgumentException("Invalid appointment status received");
+            }
+            $this->appointmentService->updateAppointmentStatus($pc_eid, $status, $session['user_id'], $session['encounter']);
+            $this->logger->debug(
+                "TeleconferenceRoomController->setAppointmentStatusAction() Updated appointment status",
+                ['pc_eid' => $pc_eid, 'status' => $status, 'authUser' => $authUser]
+            );
+            echo text(json_encode(['status' => 'success']));
+        } catch (InvalidArgumentException $exception) {
+            (new SystemLogger())->errorLogCaller($exception->getMessage(), ['trace' => $exception->getTraceAsString(),
+                'queryVars' => $queryVars]);
+            http_response_code(400);
+            echo text(json_encode(['error' => 'invalid argument sent.  Check server logs for details']));
+        } catch (AccessDeniedException $exception) {
+            (new SystemLogger())->errorLogCaller($exception->getMessage(), ['trace' => $exception->getTraceAsString(),
+                'queryVars' => $queryVars]);
+            http_response_code(403);
+            echo text(json_encode(['error' => 'Access denied to patient telehealth information.']));
+        } catch (Exception $exception) {
+            (new SystemLogger())->errorLogCaller($exception->getMessage(), ['trace' => $exception->getTraceAsString(),
+                'queryVars' => $queryVars]);
+            http_response_code(500);
+            echo text(json_encode(['error' => 'server error occurred.  Check server logs for details']));
+        }
+    }
+
+    public function getTeleHealthLaunchDataAction($queryVars)
+    {
+        try {
+            // grab the settings, grab the waiting room & grab the conference room
+            $waitingRoom = $this->renderWaitingRoom($queryVars);
+            if ($this->isPatient) {
+                $settings = $this->getPatientSettings($queryVars);
+            } else {
+                $settings = $this->getProviderSettings($queryVars);
+            }
+            $conferenceRoom = $this->renderConferenceRoom($queryVars);
+
+            $result = [
+                // waiting room has already been escaped via twig rendering
+                'waitingRoom' => $waitingRoom
+                , 'callerSettings' => textArray($settings)
+                // conference room has already been escaped via twig rendering
+                , 'conferenceRoom' => $conferenceRoom
+            ];
+            echo json_encode($result);
```

Comment:
i think need to remove json_encode above since using the textArray mechanism and letting already escaped waitingRoom and conferenceRoom unharmed.

Reply:
I don't believe I can do that as I've still got a php array here and I need to consume it via an AJAX call.  That's why I've got to do the json_encode.

Reply:
yep, all this talk of escaping has made me a bit crazy. For some reason, was thinking the json_encode was a text() :)
You are correct, keep the json_encode :)